### PR TITLE
Remove SCSSCacher and compile scss files with npm-sass

### PR DESCRIPTION
**Pull request by skjnldsv**
Replace https://github.com/nextcloud/server/pull/31743

### CODE REVIEW

File: apps/theming/css/theming.css [hunk 1]
```diff
@@ -0,0 +1,17 @@
+/* Error: Undefined variable.
```
Comment (CarlSchwan) [inline apps/theming/css/theming.css:1]: What is the goal of this file?

File: lib/composer/bamarni/composer-bin-plugin/LICENSE [hunk 2]
```diff
@@ -0,0 +1,19 @@
+Copyright (c) 2016 Bilal Amarni
```
Comment (CarlSchwan) [inline lib/composer/bamarni/composer-bin-plugin/LICENSE:1]: Don't add this, maybe we should qdd it to . gitignore

File: apps/theming/css/theming.css (hunk 1 above)
Comment (skjnldsv) [reply apps/theming/css/theming.css:1]: This one is broken and a leftover from the old theming :)

File: lib/composer/bamarni/composer-bin-plugin/LICENSE (hunk 2 above)
Comment (skjnldsv) [reply lib/composer/bamarni/composer-bin-plugin/LICENSE:1]: I'll remove it :)

**Comment by CarlSchwan** [comment]
To give a small performance overview in term of db requests

**loading dashboard page:**
- before: 19
- after: 12

**loading photo page:**
- before: 34
- after: 22

**loading files page:**
- before: 31
- after: 23

This is will a hot redis cache, will a cold cache before the numbers were always above 100 and after this patch: the number is almost the same with and without hot cache. So really nice improvement :)

File: apps/theming/css/theming.css (hunk 1 above)
Comment (artonge) [reply apps/theming/css/theming.css:1]: Will the file be removed then?

File: core/css/functions.scss [hunk 3]
```diff
@@ -27,7 +29,7 @@
  * @return string The color without #
  */
 @function remove-hash-from-color($color) {
-	$color: unquote($color);
+	// $color: string.unquote($color);
```
[the remark this answers is not in the document]
Comment (skjnldsv) [reply core/css/functions.scss:32]: It's only used by the icon-color functions, I think we could also drop those methods and the svg colour endpoint :wink:

**Comment by kesselb** [comment]
cc @MorrisJobke as we talked about removing the scss cacher a while ago :sunglasses:

**Comment by PVince81** [comment]
- [ ] BUG: tag icon missing on the files left sidebar for me:
image

I had to force refresh to get the other sidebar icons to appear in the first place, I guess this is expected

**Comment by PVince81** [comment]
- [ ] BUG: checkboxes not rendering properly in share dropdown

image

when ticked it appears fine

**Comment by PVince81** [comment]
- [ ] BUG: file picker (move/copy mode) is missing icons

image

**Comment by PVince81** [comment]
- [ ] BUG: favorite star missing on icon in file list

image

**Comment by PVince81** [comment]
I see icons missing also for other apps like activity but I assume that those apps will need to be adjusted separately anyway, we could do that after merging.

I still think we should at least fix my findings above as they are from code in the server repo

**Review by PVince81 — CHANGES_REQUESTED** [review]
See comments about missing icons, checkboxes

**Comment by skjnldsv** [comment]
Addressed @PVince81 :)
Also found some other fixups from Vanessa :rocket:

**Comment by nickvergessen** [comment]
> Conflicting files
core/css/toast.scss
dist/workflowengine-workflowengine.js
dist/workflowengine-workflowengine.js.map

**Review by PVince81 — APPROVED** [review]
:+1: looks fine now

next up: adjust all apps we know of

**Comment by CarlSchwan** [comment]
Tested again and works fine here :)

**Review by CarlSchwan — APPROVED** [review]

**Comment by skjnldsv** [comment]
Rebased, ready to get in

**Review by nickvergessen — APPROVED** [review]
Totally looked at the full diff and made sure it works in every corner case like it should

**Comment by szaimen** [comment]
🎉🎉🎉🎉🎉

**Comment by MorrisJobke** [comment]
Nice @skjnldsv! Thanks for moving this one forward 🥳

**Pull request merged**